File: Clearance/Models/Weekday.swift
//
//  Weekday.swift
//  Clearance
//
//  Days of the week, aligned with `Calendar`'s 1...7 (Sunday = 1) numbering so a
//  `Date` maps straight onto a case. Used by the weekly activity schedule.
//

import Foundation

enum Weekday: Int, CaseIterable, Identifiable, Codable, Sendable {
    case sunday = 1
    case monday
    case tuesday
    case wednesday
    case thursday
    case friday
    case saturday

    var id: Int { rawValue }

    /// The weekday for a given date in the current calendar.
    static func of(_ date: Date, calendar: Calendar = .current) -> Weekday {
        Weekday(rawValue: calendar.component(.weekday, from: date)) ?? .monday
    }

    /// Full name, e.g. "Wednesday".
    var label: String {
        switch self {
        case .sunday: "Sunday"
        case .monday: "Monday"
        case .tuesday: "Tuesday"
        case .wednesday: "Wednesday"
        case .thursday: "Thursday"
        case .friday: "Friday"
        case .saturday: "Saturday"
        }
    }

    /// Three-letter abbreviation, e.g. "Wed".
    var short: String { String(label.prefix(3)) }

    /// Cases ordered Monday → Sunday for display (most people's mental week).
    static var displayOrder: [Weekday] {
        [.monday, .tuesday, .wednesday, .thursday, .friday, .saturday, .sunday]
    }
}

File: android/app/src/main/java/com/krya1012/clearance/data/Weekday.kt
package com.krya1012.clearance.data

import java.util.Calendar
import java.util.Date

/**
 * Days of the week, aligned with `Calendar.DAY_OF_WEEK` (Sunday = 1) so a
 * `Date` maps straight onto a case — same numbering as iOS `Weekday`.
 */
enum class Weekday(val rawValue: Int) {
    SUNDAY(1),
    MONDAY(2),
    TUESDAY(3),
    WEDNESDAY(4),
    THURSDAY(5),
    FRIDAY(6),
    SATURDAY(7);

    val label: String
        get() = when (this) {
            SUNDAY -> "Sunday"
            MONDAY -> "Monday"
            TUESDAY -> "Tuesday"
            WEDNESDAY -> "Wednesday"
            THURSDAY -> "Thursday"
            FRIDAY -> "Friday"
            SATURDAY -> "Saturday"
        }

    val short: String get() = label.take(3)

    companion object {
        fun of(date: Date, calendar: Calendar = Calendar.getInstance()): Weekday {
            val cal = calendar.clone() as Calendar
            cal.time = date
            return fromRawValue(cal.get(Calendar.DAY_OF_WEEK)) ?: MONDAY
        }

        fun fromRawValue(rawValue: Int): Weekday? = entries.firstOrNull { it.rawValue == rawValue }

        /** Ordered Monday → Sunday for display (most people's mental week). */
        val displayOrder: List<Weekday> =
            listOf(MONDAY, TUESDAY, WEDNESDAY, THURSDAY, FRIDAY, SATURDAY, SUNDAY)
    }
}
